lib/solitude/api.py

```diff
@@ -849,18 +849,23 @@ def create_transaction(self, generic_buyer,
         redirect_url_onsuccess = absolutify(reverse('bango.success'))
         redirect_url_onerror = absolutify(reverse('bango.error'))
 
-        res = self.api.billing.post({
-            'pageTitle': product_name,
-            'prices': prices,
-            'transaction_uuid': transaction_uuid,
-            'seller_product_bango': provider_product['resource_uri'],
-            'redirect_url_onsuccess': redirect_url_onsuccess,
-            'redirect_url_onerror': redirect_url_onerror,
-            'icon_url': icon_url,
-            'user_uuid': user_uuid,
-            'application_size': application_size,
-            'source': source
-        })
+        try:
+            res = self.api.billing.post({
+                'pageTitle': product_name,
+                'prices': prices,
+                'transaction_uuid': transaction_uuid,
+                'seller_product_bango': provider_product['resource_uri'],
+                'redirect_url_onsuccess': redirect_url_onsuccess,
+                'redirect_url_onerror': redirect_url_onerror,
+                'icon_url': icon_url,
+                'user_uuid': user_uuid,
+                'application_size': application_size,
+                'source': source
+            })
+        except HttpClientError:
+            raise BangoTransactionCreationError(
+                'Bango transaction create failed for transaction: {tr}'
+                .format(tr=transaction_uuid))
```

Comment:
I was wondering about that, because this goes back up to tasks start_pay, now i've got celery working again I wouldn't mind giving this a full test through.